python/pyspark/sql/connect/plan.py

```diff
@@ -721,8 +721,17 @@ def plan(self, session: "SparkConnectClient") -> proto.Relation:
                     rel.join.using_columns.append(self.on)
                 else:
                     rel.join.join_condition.CopyFrom(self.to_attr_or_expression(self.on, session))
-            else:
-                rel.join.using_columns.extend(self.on)
+            elif len(self.on) > 0:
+                if isinstance(self.on[0], str):
+                    rel.join.using_columns.extend(cast(str, self.on))
+                else:
+                    merge_column = None
+                    for c in self.on:
+                        if merge_column is not None:
+                            merge_column = merge_column & c
+                        else:
+                            merge_column = c
```

Comment:
maybe `functools.reduce(lambda c1, c2: c1 & c2, self.on)`

Reply:
This is great! Applied your suggestion.